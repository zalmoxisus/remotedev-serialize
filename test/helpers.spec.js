var helpers = require('../helpers');
var mark = helpers.mark;
var extract = helpers.extract;
var extractArray = helpers.extractArray;
var refer = helpers.refer;

describe('Helpers', function () {
  it('mark', function() {
    expect(mark({ testData: 'test' }, 'testType')).toMatchSnapshot();
    expect(mark({ testData: 'test' }, 'testType', 'toString')).toMatchSnapshot();
  });

  it('extract', function() {
    expect(extract({ testData: 'test' }, 'testType')).toMatchSnapshot();
  });

  it('extractArray', function() {
    expect(extractArray([ 'Test', 'Data' ], 'testType')).toMatchSnapshot();
    expect(extractArray([ 'Test', 'Data' ], 'testType', 'toString')).toMatchSnapshot();
  });

  it('refer', function() {
    var TestClass = function(data) { return data; };
    var testInstance = new TestClass({ testData: 'test' });
    expect(refer(testInstance, 'testType', false, [TestClass])).toMatchSnapshot();
  });

});
