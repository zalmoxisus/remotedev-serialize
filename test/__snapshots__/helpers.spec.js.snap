exports[`Helpers extract 1`] = `
Object {
  "__serializedType__": "testType",
  "data": Object {
    "testData": "test",
  },
}
`;

exports[`Helpers extractArray 1`] = `
Object {
  "__serializedType__": "testType",
  "data": Array [
    "Test",
    "Data",
  ],
}
`;

exports[`Helpers extractArray 2`] = `
Object {
  "__serializedType__": "testType",
  "data": Array [
    "T",
    "e",
    "s",
    "t",
    ",",
    "D",
    "a",
    "t",
    "a",
  ],
}
`;

exports[`Helpers mark 1`] = `
Object {
  "__serializedType__": "testType",
  "data": Object {
    "testData": "test",
  },
}
`;

exports[`Helpers mark 2`] = `
Object {
  "__serializedType__": "testType",
  "data": "[object Object]",
}
`;

exports[`Helpers refer 1`] = `
Object {
  "__serializedType__": "testType",
  "data": Object {
    "testData": "test",
  },
}
`;
